posthog/api/decide.py

```diff
@@ -78,15 +80,11 @@ def get_decide(request: HttpRequest):
 
     if request.method == "POST":
         try:
-            data_from_request = load_data_from_request(request)
-            data = data_from_request["data"]
-        except (json.decoder.JSONDecodeError, TypeError):
+            data = load_data_from_request(request)
+        except RequestParsingError as error:
+            capture_exception(error)  # We still capture this on Sentry to identify actual potential bugs
             return cors_response(
-                request,
-                JsonResponse(
-                    {"code": "validation", "message": "Malformed request data. Make sure you're sending valid JSON.",},
-                    status=400,
-                ),
+                request, generate_exception_response(f"Malformed request data: {error}", code="malformed_data"),
```

Comment:
Thoughts on sending the plain error string in the response? On the one hand this is a public endpoint, on the other, code is open source anyways, and I think this is more helpful than potentially dangerous, but definitely need a second opinion.

Reply:
So in this current form this isn't a string, right? Looks like an exceptions hog err object.

Reply:
So what we generate is a JSON response that has the same format as a drf-exceptions-hog error response (for consistency). But in the error details (`{error}` in the referenced line), we print out the full Python/Django exception. For instance, if the error is that the JSON is invalid we'll send the full message telling the user where the error is (line & col), but this could be almost any type of error (e.g. passing an incorrectly encoded param, ...), so potentially could contain useful information for attackers.

Reply:
Will merge for now, but would like to still continue the conversation on this to ensure we don't have any security concerns.

Reply:
Well, `load_data_from_request` is a pure function AND open source, so I think this is OK. I don't see this particular bit of functionality facilitating an attack vector.